Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.h

```diff
@@ -87,62 +76,47 @@ namespace Terrain
             float m_sampleSpacing;
             float m_heightScale;
         };
+        
+        struct VertexPosition
+        {
+            float m_posx;
+            float m_posy;
+        };
 
-        // RPI::SceneNotificationBus overrides ...
-        void OnRenderPipelineAdded(AZ::RPI::RenderPipelinePtr pipeline) override;
-        void OnRenderPipelineRemoved(AZ::RPI::RenderPipeline* pipeline) override;
-        void OnRenderPipelinePassesChanged(AZ::RPI::RenderPipeline* renderPipeline) override;
-
-        void InitializeAtomStuff();
-        void ConfigurePipelineState(ShaderState& shaderState, bool assertOnFail);
-
-        void InitializeTerrainPatch();
+        struct VertexUv
+        {
+            float m_u;
+            float m_v;
+        };
 
-        bool InitializeRenderBuffers();
-        void DestroyRenderBuffers();
+        struct PatchData
+        {
+            AZStd::vector<VertexPosition> m_positions;
+            AZStd::vector<VertexUv> m_uvs;
+            AZStd::vector<uint16_t> m_indices;
+        };
+        
+        void Initialize();
+        void InitializeTerrainPatch(uint16_t gridSize, float gridSpacing, PatchData& patchdata);
+        bool InitializePatchModel();
 
         void ProcessSurfaces(const FeatureProcessor::RenderPacket& process);
+        
+        AZ::Outcome<AZ::Data::Asset<AZ::RPI::BufferAsset>> CreateBufferAsset(
+            const void* data, const AZ::RHI::BufferViewDescriptor& bufferViewDescriptor, const AZStd::string& bufferName);
 
         // System-level parameters
-        const float m_gridSpacing{ 1.0f };
-        const float m_gridMeters{ 32.0f };
-
-        // System-level cached reference to the Atom RHI
-        AZ::RHI::RHISystemInterface* m_rhiSystem = nullptr;
+        static constexpr float GridSpacing{ 1.0f };
+        static constexpr uint32_t GridSize{ 64 }; // number of terrain quads (vertices are m_gridSize + 1)
+        static constexpr float GridMeters{ GridSpacing * GridSize };
```

Comment:
Just a note for later - we should probably tie these settings together with our height query resolution so that our highest LOD matches the resolution of our height data.

Reply:
If I keep smoothing as an option then it'll be possible to go higher on the terrain mesh than the source data, but it should still be some multiple probably. But yeah in general I think these constants all go away and become settings in the feature processor driven from the render component.